examples/controller_test.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+"""
+Controller module for uArm Controller.
+
+This module provides a serial handler to communicate with uArm Controller Arduino MEGA 2560.
+<Event#> is a decimal value defined in ufactory.event.Event
+"""
+import sys
+import threading
+import time
+
+
+import logging
+import serial
+import serial.threaded
+
+#import heartbeat_pb2
+import line_protocol_pb2
+
+# variables
+""" registration status of R LED """
+r_LED_reg_status = False
+r_LED_on = False
+r_LED_profile_id = 333
+
+class ControllerPacketHandler(serial.threaded.Packetizer):
+    """Callback for received packet"""
+    def handle_packet(self, packet):
+        feedback = line_protocol_pb2.Feedback()
+        feedback.ParseFromString(packet)
+        #logging.info(">> asset_id: %d, message: %s", feedback.asset_id, feedback.message)
+        logging.info(">> received feedback with profile_id: %d", feedback.profile_id)
+        # if correct message received => led should be initialized
+        global r_LED_reg_status
+        global r_LED_on
```

Comment:
Yes, the `controller_test.py` is not well structured as it is only for testing the firmware.
It would be nice if we can change this such that requests can be sent dynamically via the terminal.

Reply:
>such that requests can be sent dynamically via the terminal
Hmm, usually for the testing purpose it would be better if we have automated tests with simulated pseudo TTY, but not sure if a convenient method exists for Arduino projects.

Reply:
Try the debugger with simavr and find out the virtual TTY/COM.